src/RetryMiddleware.php

```diff
@@ -130,7 +130,7 @@ public function __invoke(
         ) {
             $this->updateHttpStats($value, $requestStats);
 
-            if ($value instanceof \Exception) {
+            if ($value instanceof \Exception || $value instanceof \Error) {
```

Comment:
Maybe create a helper method for this that only checks for `\Error` if it exists.